tilelang/cuda/language/random.py

```
@@ -0,0 +1,70 @@
+    Returns
+    -------
+    state : PrimExpr
+        The random number generator state handle.
+    """
+    assert generator in ["curandStateMRG32k3a_t", "curandStatePhilox4_32_10_t", "curandStateXORWOW_t"]
+    seed = tirx.convert(seed)
+    if seq is None:
+        bx = T.get_block_binding()
+        ex = T.kernel.get_thread_extent()
+        tx = T.get_thread_binding()
+        id = tx + bx * ex
```

Ruff (0.15.21)

[error] Variable `id` is shadowing a Python builtin

(A001)